src/Raven.Server/RavenServer.cs

```diff
@@ -1563,6 +1583,22 @@ private bool TryAuthorize(RavenConfiguration configuration, Stream stream, TcpCo
                         default:
                             throw new InvalidOperationException("Unknown operation " + header.Operation);
                     }
+                case AuthenticationStatus.UnfamiliarCertificate:
+                    var info = header.AuthorizeInfo;
+                    if (info != null && info.AuthorizeAs == TcpConnectionHeaderMessage.AuthorizationInfo.AuthorizeMethod.PullReplication)
+                    {
+                        using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
+                        using (ctx.OpenReadTransaction())
+                        {
+                            var pullReplicationDefinition = ServerStore.Cluster.ReadPullReplicationDefinition(header.DatabaseName, info.AuthorizationFor, ctx);
+                            if (pullReplicationDefinition.CanAccess(certificate.Thumbprint, out var err))
```

Comment:
What if the pull replication doesn't exists?
Note that in this case we must provide the same error as if the cert doesn't exists.
What error do we get from missing db in this case, for that matter?